### PR TITLE
Add github actions ci

### PR DESCRIPTION
**Pull request by tomkennedy513**

### CODE REVIEW

File: .github/actions/pack-build/action.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      export PATH="$PATH:$(pwd)"
+      pack build ${{ inputs.tag }} \
+        --builder ${{ inputs.builder }} \
+        --env BP_GO_TARGETS="${{ inputs.bp_go_targets }}" \
+        --report-output-dir . \
+        --cache-image ${{ inputs.tag }}-cache \
+        --publish ${{ inputs.additional_pack_args }}
+      
+      mkdir images
+      digest=$(go run .github/actions/pack-build/report.go -path ./report.toml)
```
Comment (chenbh) [inline .github/actions/pack-build/action.yml:50]: Turns out `yq` can handle toml now: `yq '.image.digest' report.toml`. (I don't have an opinion which way we do it, but I just found it interesting there's finally a stream processor for toml)

File: .github/actions/imgpkg-push/action.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      data:
+        version: ${{ env.VERSION }}
+      EOF
+      
+      cat imgpkg-bundle/config/version.yml
+
+  - name: Create imagevalues.yaml
+    shell: bash
+    run: |
+      cat <<EOF > imgpkg-bundle/config/imagevalues.yml
```
[the remark this answers is not in the document]
Comment (tomkennedy513) [reply .github/actions/imgpkg-push/action.yml:52]: inagevalues.yaml is the schema that defines the values that will be provided by images.yaml or by -v if deploying locally

File: .github/actions/pack-build/action.yml (hunk 1 above)
Comment (tomkennedy513) [reply .github/actions/pack-build/action.yml:50]: Might be worth looking into

File: .github/actions/imgpkg-push/action.yml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      mkdir -p imgpkg-bundle
+      cp -r config imgpkg-bundle
+
+  - name: Extract version
+    shell: bash
+    run: |
+      [[ $GITHUB_REF =~ ^refs\/heads\/release\/(.*)$ ]] && version=${BASH_REMATCH[1]} || version=${{ github.sha }}
+      echo "VERSION=${version}" >> $GITHUB_ENV
+
+  - name: Create version yaml
```
[the remark this answers is not in the document]
Comment (tomkennedy513) [reply .github/actions/imgpkg-push/action.yml:33]: I think that's a good idea. I only did it this way because I was recreating the existing ci

**Review by chenbh — APPROVED** [review]

**Review by chenbh — APPROVED** [review]

**Pull request merged**